mParticle-Apple-SDK/Utils/NSDictionary+MPCaseInsensitive.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  NSDictionary+MPCaseInsensitive.swift
+//  mParticle-Apple-SDK
+//
+//  Created by Brandon Stalnaker on 1/29/25.
+//
+
+import Foundation
+public extension NSDictionary {
+
+    @objc func caseInsensitiveKey(_ key: String) -> String? {
+        var localKey = key
+        
+        self.allKeys.forEach { keyValue in
+            if let stringKey = keyValue as? String, stringKey.caseInsensitiveCompare(key) == .orderedSame {
+                localKey = stringKey
+            }
+        }
+        return localKey
```

Comment:
This has slightly different behavior than the original Obj-C code. In the original, it stops searching after finding the correct key, but in this code it will continue. It's not possible to use break here as it it's an actual for loop, and if you use return it will return from the closure, but I believe it will continue to loop. Instead this can be written as a for in loop:

```suggestion
        for aKey in allKeys {
            if let stringKey = aKey as? String, stringKey.caseInsensitiveCompare(key) == .orderedSame {
                return stringKey
            }
        }
        return key
```